pkg/aws/aws_cloud_test.go

```diff
@@ -0,0 +1,89 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/submariner-io/cloud-prepare/pkg/api"
+)
+
+var _ = Describe("AWS Peering", func() {
+	Context("Accept Peering", testCreateVpcPeering)
+})
+
+func testCreateVpcPeering() {
+	cloudA := newCloudTestDriver(infraID, region)
+	_ = Describe("VPC Peering", func() {
+		When("receiving a target Cloud", func() {
+			It("is an unsupported Cloud", func() {
+				invalidCloud := &fooCloud{}
+				err := cloudA.cloud.CreateVpcPeering(invalidCloud, api.NewLoggingReporter())
+				Expect(err).Should(MatchError("only AWS clients are supported"))
```

Comment:
It's better to not check for specific error messages (keeps it more backbox). If you want to distinguish errors then create specific error types. But I don't think that's necessary here.

```suggestion
				Expect(err).To(HaveOccurred())
```

Reply:
This is to make sure we're not receiving any other error as we're trying to cover all the code. What do you suggest?

Reply:
See my above comment:

> If you want to distinguish errors then create specific error types.

But as also mentioned, I think that's overkill in this case. The thing that's important is the it returns **an** error. If it tried to invoke `invalidCloud`, it would panic.